plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2007,7 +2020,22 @@ producer_plugin_impl::push_transaction( const fc::time_point& block_deadline,
       }
    }
 
+   // setting execution mode will be moved to final
+   // place when multi-threaded read-only transaction is implemented
+   auto set_execution_mode_scoped_exit = fc::make_scoped_exit([trx, &chain]{
+      if( trx->is_transient() )
+         chain.set_execution_mode( trx_execution_mode::REGULAR );
+   });
+   if( trx->is_read_only() )
+      chain.set_execution_mode(trx_execution_mode::READ_ONLY);
+   else if ( trx->is_dry_run() )
+      chain.set_execution_mode(trx_execution_mode::DRY_RUN);
    auto trace = chain.push_transaction( trx, block_deadline, max_trx_time, prev_billed_cpu_time_us, false, sub_bill );
+   if( trx->is_transient() ) {
+      chain.set_execution_mode( trx_execution_mode::REGULAR );
+      set_execution_mode_scoped_exit.cancel();
+   }
+
    auto end = fc::time_point::now();
    push_result pr;
    if( trace->except ) {
```

Comment:
I wonder if we should track read-only/dry-run trx fail-time and success-time separately than regular trxs. See `account_failures::report`.

Reply:
I think we should separate them from regular transactions. Separate stats will be useful. Will add `dry_run_trx_success_time, dry_run_trx_success_num, dry_run_trx_fail_time, dry_run_trx_fail_num; read_only_trx_success_time, read_only_trx_success_num, read_only_trx_fail_time, read_only_trx_fail_num`.

Reply:
I'm not convinced this is needed. If you do add it, seems like a separate `fc_dlog` with these details if any are `!= 0` would be good. Many nodes will not have any dry-run or read-only trxs.

Reply:
OK. We won't do it at the moment, not to add unnecessary code.